tests/cloner_test.go

```diff
@@ -75,6 +76,219 @@ var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:compo
+		By("Waiting for clone to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, targetDv3.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(targetDv3), matchFile, md5sum[:32])).To(BeTrue())
+	})
+})
+
+var _ = Describe("Validate Data Volume clone to smaller size", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+	tinyCoreIsoURL := fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs)
+
+	var (
+		sourceDv, targetDv *cdiv1.DataVolume
+		err                error
+	)
+
+	AfterEach(func() {
+		if sourceDv != nil {
+			By("Cleaning up source DV")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, sourceDv.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if targetDv != nil {
+			By("Cleaning up target DV")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, targetDv.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Should not allow cloning into a smaller sized data volume", func() {
+		By("Creating a source from a real image")
+		sourceDv = utils.NewDataVolumeWithHTTPImport("source-dv", "500Mi", tinyCoreIsoURL)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, sourceDv)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for import to be completed")
+		utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, sourceDv.Name)
+
+		By("Calculating the md5sum of the source data volume")
+		md5sum, err := f.RunCommandAndCaptureOutput(utils.PersistentVolumeClaimFromDataVolume(sourceDv), "md5sum /pvc/disk.img")
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Fprintf(GinkgoWriter, "INFO: MD5SUM for source is: %s\n", md5sum[:32])
+
+		By("Cloning from the source DataVolume to under sized target")
+		targetDv = utils.NewDataVolumeForImageCloning("target-dv", "50Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv)
+		Expect(err).To(HaveOccurred())
+		Expect(strings.Contains(err.Error(), "Target resources requests storage size is smaller than the source")).To(BeTrue())
+
+		By("Cloning from the source DataVolume to properly sized target")
+		targetDv = utils.NewDataVolumeForImageCloning("target-dv", "500Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for clone to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, targetDv.Name)
+		Expect(err).ToNot(HaveOccurred())
+		matchFile := filepath.Join(testBaseDir, "disk.img")
+		Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(targetDv), matchFile, md5sum[:32])).To(BeTrue())
+	})
+})
+
+var _ = Describe("Validate Data Volume should clone multiple clones in parallel", func() {
```

Comment:
Please add the Polarion ID as follows:
[rfe_id:CNV-1277][case_id:CNV-1899][crit:High][vendor:cnv-qe@redhat.com][level:component]